docs/frontend-system/utility-apis/02-creating.md

```diff
@@ -83,8 +84,60 @@ For illustration we make a skeleton implementation class and the API extension a
 
 The code also illustrates how the API factory declares a dependency on another utility API - the core storage API in this case. An instance of that utility API is then provided to the factory function.
 
-The resulting extension ID of the work API will be the kind `api:` followed by the plugin ID as the namespace, in this case ending up as `api:plugin.example.work`. You can now use this ID to refer to the API in app-config and elsewhere.
+The resulting extension ID of the work API will be the kind `api:` followed by the plugin ID as the namespace, in this case ending up as `api:plugin.example.work`. Check out [the naming patterns doc](../architecture/08-naming-patterns.md) for more information on how this works. You can now use this ID to refer to the API in app-config and elsewhere.
+
+## Adding configurability
+
+Here we will describe how to amend a utility API with the capability of being configured in app-config. You do this by giving a config schema to your API extension factory function. Let's make the required additions to our original work example API.
+
+```tsx title="in @internal/plugin-example"
+/* highlight-add-next-line */
+import { createSchemaFromZod } from '@backstage/frontend-plugin-api';
+
+const exampleWorkApi = createApiExtension({
+  /* highlight-add-start */
+  api: workApiRef,
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      goSlow: z.boolean().default(false),
+    }),
+  ),
+  /* highlight-add-end */
+  /* highlight-remove-next-line */
+  factory: createApiFactory({
+  /* highlight-add-next-line */
+  factory: ({ config }) => createApiFactory({
+    api: workApiRef,
+    deps: { storageApi: storageApiRef },
+    factory: ({ storageApi }) => {
+      /* highlight-add-start */
+      if (config.goSlow) {
+        /* ... */
+      }
+      /* highlight-add-end */
+    },
+  }),
+});
+```
+
+We wanted users to be able to configure a `goSlow` parameter for our API instances. So we passed in a `configSchema` to `createApiExtension` which matches that interface. This example builds it using [the zod library](https://zod.dev/). The actual config values will then be passed in a type safe manner in to the `factory` which is now a callback, wherein we can do what we wish with them. When changing to the callback form, we also had to add a top level `api: workApiRef` under `createApiExtension`.
+
+Note that while we use the word "config" here, it's _not_ the same thing as the `configApi` which gives you access to the full app-config. The config discussed here is instead the particular configuration settings given to your utility API instance. This is discussed more [in the Configuring section below](./04-configuring.md).
```

Comment:
I think it's worth explicitly calling this "extension config" and make sure we refer to it as such in a bunch of places.

Reply:
Done, that really helped